pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -195,6 +208,27 @@ func (r *ReconcileDeployment) EventBroadcaster() record.EventBroadcaster {
 	return r.evBroadcaster
 }
 
+// AddHook adds given reconcile hook to hooks list and returns the
+// an unique id that should be passed while removing the hook
+func (r *ReconcileDeployment) AddHook(h ReconcileHook) int {
```

Comment:
You don't need the id. Function pointers can be compared and thus RemoveHook can take the same argument as AddHook.

But IMHO this is all overkill. A single pointer is enough for the purpose.

Reply:
Made this simple by moving to the map.